herddb-core/src/main/java/herddb/sql/SQLRecordFunction.java

```diff
@@ -46,37 +47,81 @@
     private final Table table;
     private final List<String> columns;
     private final List<CompiledSQLExpression> expressions;
+    private final Map<String, CompiledSQLExpression> expressionsByColumnName;
 
     public SQLRecordFunction(Table table, List<Column> columns, List<CompiledSQLExpression> expressions) {
         this.table = table;
         this.columns = columns.stream().map(Column::getColumnName).collect(Collectors.toList());
         this.expressions = expressions;
+        this.expressionsByColumnName = new HashMap<>();
+        final int size = columns.size();
+        for (int i = 0; i < size; i++) {
+            expressionsByColumnName.put(this.columns.get(i), expressions.get(i));
+        }
     }
 
     public SQLRecordFunction(List<String> columns, Table table, List<CompiledSQLExpression> expressions) {
         this.table = table;
         this.columns = columns;
         this.expressions = expressions;
+        this.expressionsByColumnName = new HashMap<>();
+        final int size = columns.size();
+        for (int i = 0; i < size; i++) {
+            expressionsByColumnName.put(this.columns.get(i), expressions.get(i));
+        }
     }
 
     @Override
     public byte[] computeNewValue(Record previous, StatementEvaluationContext context, TableContext tableContext) throws StatementExecutionException {
         try {
-            Map<String, Object> res = previous != null ? new HashMap<>(previous.toBean(table)) : new HashMap<>(table.columns.length);
+            byte[] oldRes = _computeNewValueOld(previous, context);
+            byte[] newRes = _computeNewValue(previous, context, tableContext);
+            if (!Arrays.equals(oldRes, newRes)) {
```

Comment:
Computing both of them don't we loose the patch benefits?
It should be under some assertion or debug mode to avoid computations in standard runtime

Reply:
it is only debug during tests, to be sure that the new algorithm serializes data exactly as the previous one